### PR TITLE
Prac 06 Code Review

### PR DESCRIPTION
**Pull request by Nathan-Dachille**
Good evening, @alexchristie12, 

Can you please review my prac 6 code?

Cheers Nathan

### CODE REVIEW

File: prac_06/car.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    def add_fuel(self, amount):
+        """Add amount to the car's fuel."""
+        self.fuel += amount
+
+    def drive(self, distance):
+        """Drive the car a given distance.
+
+        Drive given distance if car has enough fuel
+        or drive until fuel runs out return the distance actually driven.
```
Comment (al-c7) [inline prac_06/car.py:27]: Sentence structure here isn't amazing. It gets the point across but does it read very well?

File: prac_06/programming_language.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        self.name = name
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def __str__(self):
+        return f"{self.name}, {self.typing} Typing, Reflection={self.reflection}, First appeared in {self.year}"
+
+    def is_dynamic(self):
+        if self.typing.lower() == "dynamic":
```
Comment (al-c7) [inline prac_06/programming_language.py]: Do we need an `if` statement here? Will `return self.typing.lower() == dynamic` have the same effect?

File: prac_06/languages.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+
+python = ProgrammingLanguage("Python", "Dynamic", True, 1991)
+ruby = ProgrammingLanguage("Ruby", "Dynamic", True, 1995)
+visual_basic = ProgrammingLanguage("Visual Basic", "Static", False, 1991)
+print(python)
+
+languages = [python, ruby, visual_basic]
+
+for language in languages:
```
Comment (al-c7) [inline prac_06/languages.py:18]: Have you forgotten the `"The dynamically typed languages are: "` string here?

File: prac_06/guitars.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+Estimated Time: 30min
+Actual Time: 41min
+"""
+
+from prac_06.guitar import Guitar
+
+print("My Guitars!")
+name = input("Name: ")
+guitars = []
+while name != "":
```
Comment (al-c7) [inline prac_06/guitars.py:11]: The implementation here is good, especially with how you handle the exiting of the function. But is the error checking required? Is it explicitly stated in the requirements?

**Review by al-c7 — APPROVED** [review]
Overall a good implementation for each task, with only a few minor hiccups here and there. Well done.

File: prac_06/car.py (hunk 1 above)
Comment (Nathan-Dachille) [reply prac_06/car.py:27]: Thanks for the feedback, I believe it is like that in the template though.

File: prac_06/guitars.py (hunk 4 above)
Comment (Nathan-Dachille) [reply prac_06/guitars.py:11]: Good point; I had assumed that it was good practice to error check input.

File: prac_06/languages.py (hunk 3 above)
Comment (Nathan-Dachille) [reply prac_06/languages.py:18]: Good catch... Fixed.

File: prac_06/programming_language.py (hunk 2 above)
Comment (Nathan-Dachille) [reply prac_06/programming_language.py]: Noted and fixed.

**Pull request merged**